### PR TITLE
Simplify error handling in the binary

### PR DESCRIPTION
**Pull request by neithernut**
We have a lot of ugly stuff in the binary, just for the sake of error handling. This PR aims to make this less painful.

### CODE REVIEW

**Comment by neithernut** [comment]
The idea is to make use of the new extensions in the subcommand implementations. There, it is absolutely valid to simply abort in most cases. Maybe we'll eliminate `try_or_1!` in favour for `unwrap_or_abort()`. The latter is obviously better for chaining stuff.

File: src/abort.rs [hunk 1]
```diff
@@ -64,3 +59,35 @@ impl<I, V, E> IteratorExt<I, V, E> for I
...
+{
+    /// Just like a regulat unwrap() except it performs proper logging
+    ///
+    /// Returns the contained value or aborts the program, logging the error.
+    ///
+    fn unwrap_or_abort(self) -> V;
+}
+
+impl<V, E> Abortable<V> for Result<V, E>
+    where E: Debug
```
Comment (matthiasbeyer) [inline src/abort.rs]: I'm not that sure here, but as we're using `error_chain` I'd rather use the actual error type and `error_chain`s tracing functionality and not the `error!(...)` logging macro...

Comment (neithernut) [inline src/abort.rs]: The `ErrorChain` trait implements `Display`. The documentation states that the entire chain will be printed in such cases. I assumed that `error!()` will do the right thing in such cases.

Comment (neithernut) [inline src/abort.rs]: In any case, I'd keep this implementation if possible and supplement a specialization. We also have errors we get straight from `git2`.

Comment (neithernut) [inline src/abort.rs]: Mh... `rust` complains about conflicting implementations.

**Comment by neithernut** [comment]
I'll rebase this branch on the latest master and make use of the new traits. After I'm done we can evaluate how much pain it would be to only support `ChainedError`s with the new type.

**Comment by neithernut** [comment]
I only applied changes which introduced some benefit imo so far, e.g. I didn't eliminate `try_or_1!` completely.

**Comment by neithernut** [comment]
Looking at subcommand implementations, maybe it makes sense to eliminate `try_or_1!` completely in favour for `unwrap_or_abort()` and have the subcommands return "void" instead of an "i32". It would be much cleaner imo. @matthiasbeyer?

**Comment by matthiasbeyer** [comment]
> Looking at subcommand implementations, maybe it makes sense to eliminate `try_or_1!` completely in favour for `unwrap_or_abort()` and have the subcommands return "void" instead of an "i32". It would be much cleaner imo.

Yup.

**Comment by neithernut** [comment]
Most commit messages will be rewritten in the squash

**Comment by neithernut** [comment]
Blocks #104.

**Comment by neithernut** [comment]
Blocks #105.

**Comment by neithernut** [comment]
@matthiasbeyer Are you going to review this anytime soon?

File: src/abort.rs [hunk 2]
```diff
@@ -64,3 +59,35 @@ impl<I, V, E> IteratorExt<I, V, E> for I
...
+        AbortingIter::from(self.unwrap_or_abort())
+    }
+}
+
+
+/// Extension trait for convenient abortion in case of errors
+///
+pub trait Abortable<V>
+{
+    /// Just like a regulat unwrap() except it performs proper logging
```
Comment (matthiasbeyer) [inline src/abort.rs]: typo `regulat`

**Pull request merged**